app/models/manageiq/providers/vmware/infra_manager/vm.rb

```diff
@@ -31,6 +31,10 @@ def validate_migrate
     validate_supported
   end
 
+  def validate_publish
+    validate_unsupported("Publish VM")
```

Comment:
This is probably wrong.. it's supposed to fail in `ManageIQ::Providers::Microsoft::InfraManager::Vm` and `ManageIQ::Providers::Redhat::InfraManager::Vm`, not here..